modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -0,0 +1,582 @@
+            CV_Assert(attribute_proto.ints_size() == 4);
+            lp.set("pad_h", saturate_cast<int32_t>(attribute_proto.ints(0)));
+            lp.set("pad_w", saturate_cast<int32_t>(attribute_proto.ints(1)));
+            // push pad_b and pad_r for compute ceil_mode
+            lp.set("pad_b", saturate_cast<int32_t>(attribute_proto.ints(2)));
+            lp.set("pad_r", saturate_cast<int32_t>(attribute_proto.ints(3)));
+        }
+        else if(attribute_name == "auto_pad")
+        {
+            if (attribute_proto.s() == "SAME_UPPER" || attribute_proto.s() == "SAME_LOWER") {
+                lp.set("pad_mode",  "SAME");
+            }
+            else if (attribute_proto.s() == "VALID") {
+                lp.set("pad_mode", "VALID");
+            }
+        }
+        else if(attribute_name == "dilations")
+        {
+            CV_Assert(attribute_proto.ints_size() == 2);
+            lp.set("dilation_h",  saturate_cast<int32_t>(attribute_proto.ints(0)));
+            lp.set("dilation_w",  saturate_cast<int32_t>(attribute_proto.ints(1)));
+        }
+        else if (attribute_proto.has_i())
+        {
+             ::google::protobuf::int64 src = attribute_proto.i();
+            if (src < std::numeric_limits<int32_t>::min() || src > std::numeric_limits<int32_t>::max())
+                CV_Error(Error::StsOutOfRange, "Input is out of OpenCV 32S range");
+            else
+                lp.set(attribute_name, saturate_cast<int32_t>(src));
+        }
+        else if (attribute_proto.has_f())
+        {
+            lp.set(attribute_name, attribute_proto.f());
+        }
+        else if (attribute_proto.has_s())
+        {
+            lp.set(attribute_name, attribute_proto.s());
+        }
+        else if (attribute_proto.floats_size() > 0)
+        {
+            lp.set(attribute_name, DictValue::arrayReal(
+                (float*)attribute_proto.mutable_floats(), attribute_proto.floats_size()));
+        }
+        else if (attribute_proto.ints_size() > 0)
+        {
+            const ::google::protobuf::RepeatedField< ::google::protobuf::int64 >& src = attribute_proto.ints();
+            std::vector<int32_t> dst(attribute_proto.ints_size());
+
+            for (int j = 0; j < dst.size(); j++) {
+                if (src[j] < std::numeric_limits<int32_t>::min() || src[j] > std::numeric_limits<int32_t>::max()) {
+                    CV_Error(Error::StsOutOfRange, "Input is out of OpenCV 32S range");
+                }
+                dst[j] = saturate_cast<int32_t>(src[j]);
+            }
+
+                lp.set(attribute_proto.name(), DictValue::arrayInt(
+                    &dst[0], attribute_proto.ints_size()));
+        }
+        else if (attribute_proto.has_t())
+        {
+            ::opencv_onnx::TensorProto tensor = attribute_proto.t();
+            Mat blob = getMatFromTensor(tensor);
+            lp.blobs.push_back(blob);
+        }
+        else if (attribute_proto.has_t() || attribute_proto.has_g() ||
```

Comment:
For now we can manage `attribute_proto.has_t()` (see above).

Reply:
done